use-cases/virgo/trainer.py

```diff
@@ -108,26 +110,86 @@ def create_model_loss_optimizer(self) -> None:
+            dataset=train_dataset,
+            batch_size=self.config.batch_size,
+            num_workers=self.config.num_workers_dataloader,
+            pin_memory=self.config.pin_gpu_memory,
+            generator=self.torch_rng,
+            shuffle=self.config.shuffle_train,
+            collate_fn=self.custom_collate
+        )
+        if validation_dataset is not None:
+            self.validation_dataloader = self.strategy.create_dataloader(
+                dataset=validation_dataset,
+                batch_size=self.config.batch_size,
+                num_workers=self.config.num_workers_dataloader,
+                pin_memory=self.config.pin_gpu_memory,
+                generator=self.torch_rng,
+                shuffle=self.config.shuffle_validation,
+                collate_fn=self.custom_collate
+            )
+        if test_dataset is not None:
+            self.test_dataloader = self.strategy.create_dataloader(
+                dataset=test_dataset,
+                batch_size=self.config.batch_size,
+                num_workers=self.config.num_workers_dataloader,
+                pin_memory=self.config.pin_gpu_memory,
+                generator=self.torch_rng,
+                shuffle=self.config.shuffle_test,
+                collate_fn=self.custom_collate
+            )
+
+    def custom_collate(self, batch):
+        """
+        Custom collate function to concatenate input tensors along their first dimension.
+        """
+        # Some batches contain None values, if any files from the dataset did not match the criteria
+        # (i.e. three auxilliary channels)
+        batch = list(filter(lambda x: x is not None, batch))
+        batch = torch.cat(batch)
+
+        return batch
+
     def train(self):
+        # Start the timer for profiling
+        st = timer()
         # uncomment all lines relative to accuracy if you want to measure
         # IOU between generated and real spectrograms.
         # Note that it significantly slows down the whole process
         # it also might not work as the function has not been fully
         # implemented yet
 
+        if self.strategy.is_main_worker:
+            print('TIMER: broadcast:', timer()-st, 's')
+            print('\nDEBUG: start training')
+            print('--------------------------------------------------------')
+            nnod = os.environ.get('SLURM_NNODES', 'unk')
+            s_name = f"{os.environ.get('DIST_MODE', 'unk')}-torch"
+            epoch_time_tracker = EpochTimeTracker(
+                series_name=s_name,
+                csv_file=f"epochtime_{s_name}_{nnod}N.csv"
+            )
         loss_plot = []
         val_loss_plot = []
         acc_plot = []
         val_acc_plot = []
         best_val_loss = float('inf')
+
         for epoch in tqdm(range(self.num_epochs)):
+            lt = timer()
             # itwinai - IMPORTANT: set current epoch ID
             self.set_epoch(epoch)
-
+            t_list = []
```

Comment:
I would rename this variable. Not quite sure what this is supposed to mean. Maybe "time_list"?